test/fabrik.VirtualHostAgent.spec.js

```diff
@@ -0,0 +1,196 @@
+      requestStub
+        .withArgs({
+          method: 'DELETE',
+          url: createUrl(pathname),
+          auth: auth,
+          body: body
+        }, expectedStatus)
+        .returns(Promise.resolve(response));
+      requestStub
+        .withArgs({
+          method: 'GET',
+          url: createUrl(pathname),
+          auth: auth
+        }, expectedStatus)
+        .returns(Promise.resolve(response));
+      requestStub
+        .withArgs({
+          method: 'GET',
+          url: createUrl(pathname),
+          auth: auth,
+          json: false
+        }, expectedStatus)
+        .returns(Promise.resolve({
+          body: _
+            .chain(logs)
+            .map(JSON.stringify)
+            .join('\n')
+            .value(),
+        }));
+    });
+
+    afterEach(function () {
+      requestStub.restore();
+    });
+
+    describe('#createVirtualHost', function () {
+      before(function () {
+        pathname = `tenants/${instanceId}`;
+        expectedStatus = 200;
+        _.set(body, 'parameters', {});
+      });
+
+      after(function () {
+        _.unset(body, 'parameters');
+      });
+
+      it('returns a JSON object', function () {
+        return agent
+          .createVirtualHost(ips, instanceId)
+          .then(() => {
+            expect(requestStub).to.have.been.calledTwice;
+          });
+      });
+    });
+
+    describe('#deleteVirtualHost', function () {
+      before(function () {
+        pathname = `tenants/${instanceId}`;
+        expectedStatus = 204;
+        _.set(body, 'parameters', {});
+      });
+
+      it('returns a JSON object', function () {
+        return agent
+          .deleteVirtualHost(ips, instanceId)
+          .then(() => {
+            expect(requestStub).to.have.been.calledTwice;
+          });
+      });
+
+    });
+
+    describe('#createCredentials', function () {
+      before(function () {
+        pathname = `tenants/${instanceId}/credentials`;
+        expectedStatus = 200;
+        _.set(body, 'parameters', parameters);
+      });
+
+      after(function () {
+        _.unset(body, 'parameters');
+      });
+
+      it('returns a JSON object', function () {
+        return agent
+          .createCredentials(ips, instanceId, parameters)
+          .then(body => {
+            expect(body).to.equal(response.body);
+            expect(requestStub).to.have.been.calledTwice;
+          });
+      });
+    });
+
+    describe('#deleteCredentials', function () {
+      before(function () {
+        pathname = `tenants/${instanceId}/credentials`;
+        expectedStatus = 204;
+        _.set(body, 'credentials', credentials);
+      });
+
+      after(function () {
+        _.unset(body, 'credentials');
+      });
+
+      it('returns a JSON object', function () {
+        return agent
+          .deleteCredentials(ips, instanceId, credentials)
+          .then(() => {
+            expect(requestStub).to.have.been.calledTwice;
```

Comment:
same as [above](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/111/files#r157937712)

Reply:
An assertions are added that chacks that the first call is GET and second is DELETE.